pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -73,12 +74,33 @@ public static void setUserConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, S
     propertyStore.set(constructPropertyStorePathForUserConfig(username), znRecord, AccessOption.PERSISTENT);
   }
 
+  @Deprecated
   public static void setTableConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
       ZNRecord znRecord) {
     propertyStore.set(constructPropertyStorePathForResourceConfig(tableNameWithType), znRecord,
         AccessOption.PERSISTENT);
   }
 
+  /**
+   * Update table config with an expected version. This is to avoid race condition for table config update issued by
+   * multiple clients, especially when update configs in a programmatic way.
+   * The typical usage is to read table config, apply some changes, then update it.
+   *
+   * @return true if update is successful.
+   */
+  public static boolean setTableConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
```

Comment:
Don't enforce the expected version. IMO we can probably add 3 methods:
1. Set with expected version
2. Set without expected version (always override)
3. Create (fail when it already exists)

Reply:
done.